### PR TITLE
difficulty_vote: prevent voting same difficulty.

### PR DESCRIPTION
**Pull request by PlayerNoon**
Some players tend to vote same difficulty many time in short period of time. Each vote print same messages in chat. This code prevent printing in chat message if the current vote is same as previous.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by TheBigZet** [comment]
How about moving `set_difficulty()` and `difficulty_gui_all()` inside the `if`, and adding `else` with `player.print("You already voted for this difficulty")`?

File: maps/biter_battles_v2/difficulty_vote.lua [hunk 1]
```diff
@@ -155,10 +155,14 @@ local function on_gui_click(event)
         return
     end
 	
-	game.print(player.name .. " has voted for " .. difficulties[i].name .. " difficulty!", difficulties[i].print_color)
-	global.difficulty_player_votes[player.name] = i
-	set_difficulty()
-	difficulty_gui_all()
+	if not (global.difficulty_player_votes[player.name] == i) or global.difficulty_player_votes[player.name] == nil then
```
Comment (developer-8) [inline maps/biter_battles_v2/difficulty_vote.lua:158]:
```suggestion
	if global.difficulty_player_votes[player.name] ~= i then
```
Can it be done like this, or is there some catch with nil?

**Review by Ragnarok77-factorio — APPROVED** [review]
Tested it, OK, didn't find any issue.
For code, nothing more to add.

**Comment by developer-8** [comment]
@PlayerNoon  Please reopen

**Review by TheBigZet — APPROVED** [review]

**Pull request merged**